### PR TITLE
Add missing content to cli-create.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Add offline mode and updating sections from legacy documentation.

### CODE REVIEW

File: docs/tools/CLI/cli-create.md [hunk 1]
```diff
@@ -108,9 +108,59 @@ $ mbed compile -t ARM -m K64F --source project2 --source mbed-os --build BUILD/p
 
 Find more details on the `--source` switch in the [build rules documentation](../reference/mbed-os-build-rules.html).
 
-### Updating to an upstream version
+### Updating programs and libraries
 
-#### Updating a program
+You can update programs and libraries on your local machine so that they pull in changes from the remote sources (Git or Mercurial).
```
Comment (theotherjimmy) [inline docs/tools/CLI/cli-create.md]: "pull in changes" sounds very colloquial, and may not mean much in this context. Perhaps "update to the latest released version" would be a better replacement.

File: docs/tools/CLI/cli-create.md [hunk 2]
```diff
@@ -108,9 +108,59 @@ $ mbed compile -t ARM -m K64F --source project2 --source mbed-os --build BUILD/p
...
+
+As with any Mbed CLI command, `mbed update` uses the current directory as a working context. Before calling `mbed update`, you should change your working directory to the one you want to update. For example, if you're updating `mbed-os`, use `cd mbed-os` before you begin updating.
+
+<span class="tips">**Tip:** Synchronizing library references: Before triggering an update, you may want to synchronize any changes that you've made to the program structure by running mbed sync, which updates the necessary library references and removes the invalid ones.</span>
```
Comment (theotherjimmy) [inline docs/tools/CLI/cli-create.md]:
```suggestion
<span class="tips">**Tip:** Synchronizing library references: Before triggering an update, you may want to synchronize any changes that you've made to the program structure by running `mbed sync`, which updates the necessary library references and removes the invalid ones.</span>
```

File: docs/tools/CLI/cli-create.md [hunk 3]
```diff
@@ -108,9 +108,59 @@ $ mbed compile -t ARM -m K64F --source project2 --source mbed-os --build BUILD/p
...
+
+#### Protection against overwriting local changes
+
+The update command fails if there are changes in your program or library that `mbed update` could overwrite. This is by design. Mbed CLI does not run operations that would result in overwriting uncommitted local changes. If you get an error, take care of your local changes (commit or use one of the options below), and then rerun `mbed update`.
```
Comment (theotherjimmy) [inline docs/tools/CLI/cli-create.md]: What options below?

**Review by theotherjimmy — CHANGES_REQUESTED** [review]

File: docs/tools/CLI/cli-create.md (hunk 1 above)
Comment (AnotherButler) [reply docs/tools/CLI/cli-create.md]: Done

File: docs/tools/CLI/cli-create.md (hunk 3 above)
Comment (AnotherButler) [reply docs/tools/CLI/cli-create.md]: Removed

**Review by theotherjimmy — APPROVED** [review]

**Pull request merged**